node/src/Cardano/Node/API.hs

```diff
@@ -424,46 +423,50 @@ v1SyncPercentage nodeHeight walletHeight =
                 / max 1.0 (fromIntegral nd)
                 ) * 100.0
 
+
 -- | Get the difference between NTP time and local system time, nothing if the
 -- NTP server couldn't be reached in the last 30min.
 --
 -- Note that one can force a new query to the NTP server in which case, it may
 -- take up to 30s to resolve.
---
--- Copied from Cardano.Wallet.Kernel.NodeStateAdapter
 defaultGetNtpDrift
-    :: TVar NtpStatus
+    :: MonadIO m
+    => TVar NtpStatus
     -> ForceNtpCheck
-    -> IO TimeInfo
-defaultGetNtpDrift tvar ntpCheck = mkTimeInfo <$> case ntpCheck of
-    ForceNtpCheck -> do
-        forceNtpCheck
-        getNtpOffset blockingLookupNtpOffset
-    NoNtpCheck ->
+    -> m TimeInfo
+defaultGetNtpDrift tvar ntpCheckBehavior =
+    if (ntpCheckBehavior == ForceNtpCheck) then
+        forceNtpCheck >> getNtpOffset blockingLookupNtpOffset
+    else
         getNtpOffset nonBlockingLookupNtpOffset
   where
+    forceNtpCheck :: MonadIO m => m ()
     forceNtpCheck =
         atomically $ writeTVar tvar NtpSyncPending
 
+    getNtpOffset :: MonadIO m => (NtpStatus -> STM TimeInfo) -> m TimeInfo
     getNtpOffset lookupNtpOffset =
-        atomically (readTVar tvar >>= lookupNtpOffset)
-
-    mkTimeInfo :: Maybe NtpOffset -> TimeInfo
-    mkTimeInfo =
-        TimeInfo . fmap (mkLocalTimeDifference . toMicroseconds)
+        atomically $ (readTVar tvar >>= lookupNtpOffset)
 
+    -- Lookup NtpOffset from an NTPStatus in a non-blocking manner
+    --
+    -- i.e. Returns immediately with 'Nothing' if the NtpSync is pending.
```

Comment:
This comment could be updated. It's not `Nothing` now, it's `TimeInfoPending`.

Reply:
Indeed.

Reply:
Fixed, cf: https://github.com/input-output-hk/cardano-sl/compare/d9758eb7f47bd8bce6dc697c67f915ba52e5129b..c2d220f26ca2d59629162d42a87e4a9f3d8c89a2